resources/sts/4.14/hypershift/sts_extended_hcp_support_permission_policy.json

```diff
@@ -62,49 +137,300 @@
                 "tiros:GetQueryAnswer",
-                "tiros:GetQueryExplanation",
+                "tiros:GetQueryExplanation"
+            ],
+            "Resource": "*"
+        },
+        {
+            "Sid": "NetworkVerifier1",
+            "Effect": "Allow",
+            "Action": [
+                "ec2:CreateTags"
+            ],
+            "Resource": "arn:aws:ec2:*:*:instance/*",
+            "Condition":{
+                "StringEquals":{
+                    "ec2:CreateAction":"RunInstances",
+                    "aws:RequestTag/red-hat-managed": "true"
+                },
+                "ForAllValues:StringEquals": {
+                    "aws:TagKeys": "red-hat-managed"
+                }
+            }
+        },
+        {
+            "Sid": "NetworkVerifier2",
+            "Effect": "Allow",
+            "Action": [
                 "ec2:RunInstances",
                 "ec2:DescribeInstanceTypes",
-                "ec2:CreateTags",
                 "ec2:GetConsoleOutput"
             ],
+            "Resource": "arn:aws:ec2:*:*:instance/*",
+            "Condition": {
+                "StringEquals": {
+                    "aws:ResourceTag/red-hat-managed": "true"
+                }
+            }
+        },
+        {
+            "Sid": "Ip",
+            "Effect": "Allow",
+            "Action": [
+                "ec2:DescribeCoipPools",
+                "ec2:DescribeIpv6Pools",
+                "ec2:DescribePublicIpv4Pools",
+                "ec2:GetAssociatedIpv6PoolCidrs"
+            ],
+            "Resource": "*"
+        },
+        {
+            "Sid": "InternetGateway",
+            "Effect": "Allow",
+            "Action": [
+                "ec2:DescribeEgressOnlyInternetGateways",
+                "ec2:DescribeInternetGateways"
+            ],
+            "Resource": "*"
+        },
+        {
+            "Sid": "Describetag",
+            "Effect": "Allow",
+            "Action": [
+               "ec2:DescribeTags" 
+            ],
+            "Resource": "*"
+        },
+        {
+            "Sid": "PrefixLists",
+            "Effect": "Allow",
+            "Action": [
+                "ec2:DescribePrefixLists",
+                "ec2:DescribeManagedPrefixLists"
+            ],
+            "Resource": "*"
+        },
+        {
+            "Sid": "PrefixListsentries",
+            "Effect": "Allow",
+            "Action": [
+                "ec2:GetManagedPrefixListEntries"
+            ],
+            "Resource": "arn:aws:ec2:*:*:prefix-list/*"
+        },
+        {
+            "Sid": "Image",
+            "Effect": "Allow",
+            "Action": [
+                "ec2:DescribeImages"
+            ],
+            "Resource": "*"
+        },
+        {
+            "Sid": "ImagaeAttibute",
+            "Effect": "Allow",
+            "Action": [
+                "ec2:DescribeImageAttribute"
+            ],
+            "Resource": "arn:aws:ec2:*:*:image/*"
```

Comment:
Any way these can be restricted more? Do the images all have red-hat-managed: true